assembly/src/parser/instructions_with_labels.rs

```diff
@@ -87,229 +97,372 @@ pub enum InstructionsWithLabels {
+        prover_only: bool,
     },
     Addi {
         dst: Slot,
         src1: Slot,
         imm: Immediate,
+        prover_only: bool,
     },
     Or {
         dst: Slot,
         src1: Slot,
         src2: Slot,
+        prover_only: bool,
     },
     Ori {
         dst: Slot,
         src1: Slot,
         imm: Immediate,
+        prover_only: bool,
     },
     Sub {
         dst: Slot,
         src1: Slot,
         src2: Slot,
+        prover_only: bool,
     },
     Sle {
         dst: Slot,
         src1: Slot,
         src2: Slot,
+        prover_only: bool,
     },
     Slei {
         dst: Slot,
         src: Slot,
         imm: Immediate,
+        prover_only: bool,
     },
     Sleu {
         dst: Slot,
         src1: Slot,
         src2: Slot,
+        prover_only: bool,
     },
     Sleiu {
         dst: Slot,
         src: Slot,
         imm: Immediate,
+        prover_only: bool,
     },
     Slt {
         dst: Slot,
         src1: Slot,
         src2: Slot,
+        prover_only: bool,
     },
     Slti {
         dst: Slot,
         src: Slot,
         imm: Immediate,
+        prover_only: bool,
     },
     Sltu {
         dst: Slot,
         src1: Slot,
         src2: Slot,
+        prover_only: bool,
     },
     Sltiu {
         dst: Slot,
         src: Slot,
         imm: Immediate,
+        prover_only: bool,
     },
     Sll {
         dst: Slot,
         src1: Slot,
         src2: Slot,
+        prover_only: bool,
     },
     Srl {
         dst: Slot,
         src1: Slot,
         src2: Slot,
+        prover_only: bool,
     },
     Sra {
         dst: Slot,
         src1: Slot,
         src2: Slot,
+        prover_only: bool,
     },
     Andi {
         dst: Slot,
         src1: Slot,
         imm: Immediate,
+        prover_only: bool,
     },
     And {
         dst: Slot,
         src1: Slot,
         src2: Slot,
+        prover_only: bool,
     },
     Muli {
         dst: Slot,
         src1: Slot,
         imm: Immediate,
+        prover_only: bool,
     },
     Mul {
         dst: Slot,
         src1: Slot,
         src2: Slot,
+        prover_only: bool,
     },
     Mulu {
         dst: Slot,
         src1: Slot,
         src2: Slot,
+        prover_only: bool,
     },
     Mulsu {
         dst: Slot,
         src1: Slot,
         src2: Slot,
+        prover_only: bool,
     },
     Srli {
         dst: Slot,
         src1: Slot,
         imm: Immediate,
+        prover_only: bool,
     },
     Slli {
         dst: Slot,
         src1: Slot,
         imm: Immediate,
+        prover_only: bool,
     },
     Srai {
         dst: Slot,
         src1: Slot,
         imm: Immediate,
+        prover_only: bool,
+    },
+    Alloci {
+        dst: Slot,
+        imm: Immediate,
+    },
+    Allocv {
+        dst: Slot,
+        src: Slot,
     },
     Ret,
 }
 
+impl InstructionsWithLabels {
+    fn prover_only(&self) -> bool {
+        use InstructionsWithLabels::*;
+        match self {
```

Comment:
From my understanding, only the `ALLOCI` / `ALLOCV` instructions are prover only (and soon the Groestl-related instructions for recursive proving). If that is correct, I think it'd be best to make it explicit, instead of making every instruction possibly `prover_only`?  I may be missing a nuance here though